client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -502,6 +503,13 @@ export class DomainDetailsForm extends PureComponent {
 		}
 	}
 
+	// We want to cache the functions to avoid triggering unecessary rerenders
+	getInputRefCallback( name ) {
+		return this.inputRefCallbacks[ name ] ||
+			( this.inputRefCallbacks[ name ] = ( el ) =>
+				this.inputRefs[ name ] = el );
```

Comment:
Aside: I realize the OR shortcut construct is extremely common. I still don't like it. ;)